src/providers/structure.ts

```diff
@@ -40,8 +46,50 @@ export class SectionNodeProvider implements vscode.TreeDataProvider<Section> {
     }
 
     update() {
-        this.refresh()
-        this._onDidChangeTreeData.fire(undefined)
+        this.refresh().finally(() => {this._onDidChangeTreeData.fire(undefined)})
+    }
+
+    async parseBibTeX() {
+        const document = vscode.window.activeTextEditor?.document
+        if (!document) {
+            return
+        }
+        const ast = await this.extension.pegParser.parseBibtex(document?.getText() || '').catch((error) => {
```

Comment:
No need to use `document?.` as we already check that `document` is well defined above. I guess calling `this.extension.pegParser.parseBibtex(document.getText())` would just be fine.